### PR TITLE
feature: selinux for cri manager

### PR DESCRIPTION
**Pull request by YaoZengzeng**
Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### Ⅲ. Describe how you did it

With this PR, we could enable one of the security context --- SELinux, if it's configured in CRI.

### CODE REVIEW

**Comment by YaoZengzeng** [comment]
@allencloud PTAL

File: daemon/mgr/container_utils.go [hunk 1]
```diff
@@ -103,39 +104,46 @@ func (mgr *ContainerManager) generateName(id string) string {
...
+			return fmt.Errorf("invalid --security-opt %s: must be in format of key=value", securityOpt)
+		}
+		key, value := fields[0], fields[1]
+		switch key {
+		// TODO: handle other security options.
+		case "apparmor":
+			meta.AppArmorProfile = value
+		case "seccomp":
+			meta.SeccompProfile = value
+		case "no-new-privileges":
```
Comment (allencloud) [inline daemon/mgr/container_utils.go]: prevent processes in container to gain new privileges via the `--security-opt=no-new-privileges` flag

just attach a record which may do us a favour.

File: daemon/mgr/container_utils.go [hunk 2]
```diff
@@ -103,39 +104,46 @@ func (mgr *ContainerManager) generateName(id string) string {
...
+			noNewPrivileges, err := strconv.ParseBool(value)
+			if err != nil {
+				return fmt.Errorf("invalid --security-opt: %q", securityOpt)
+			}
+			meta.NoNewPrivileges = noNewPrivileges
+		case "label":
+			labelOpts = append(labelOpts, value)
+		default:
+			return fmt.Errorf("invalid type %s in --security-opt %s: unknown type from apparmor and seccomp", key, securityOpt)
```
Comment (allencloud) [inline daemon/mgr/container_utils.go]: With the code change, I do not think this line is correct any longer. 

> unknown type from apparmor and seccomp

except `apparmor` and` seccomp`, it seems that we should add `no-new-privileges` and `label`. Right? @YaoZengzeng

File: daemon/mgr/container_utils.go [hunk 3]
```diff
@@ -103,39 +104,46 @@ func (mgr *ContainerManager) generateName(id string) string {
...
-	key, value := fields[0], fields[1]
-	switch key {
-	// TODO: handle other security options.
-	case "apparmor":
-		meta.AppArmorProfile = value
-	case "seccomp":
-		meta.SeccompProfile = value
-	case "no-new-privileges":
-		noNewPrivileges, err := strconv.ParseBool(value)
+	if labelOpts != nil {
```
Comment (allencloud) [inline daemon/mgr/container_utils.go]: We can judge `if labelOpts == nil ` first to make return fast to simplify code and reduce ident.

```
if labelOpts == nil {
    return nil
}
meta.ProcessLabel, meta.MountLabel, err = label.InitLabels(labelOpts)
if err != nil{
    return fmt.Errorf("failed to init labels: %v", err)
}
return nil
```

**Comment by YaoZengzeng** [comment]
@allencloud Updated.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**